File: app/schemas/invitation.py
import uuid
from datetime import datetime

from pydantic import BaseModel, EmailStr

from app.db.enums import UserRole


class InvitationCreateRequest(BaseModel):
    email: EmailStr
    role: UserRole


class InvitationResponse(BaseModel):
    id: uuid.UUID
    email: str
    role: UserRole
    is_used: bool
    created_at: datetime
    expires_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/invitation.ts
import type { UserRole } from "./auth";

export interface InvitationCreateRequest {
  email: string;
  role: UserRole;
}

export interface InvitationResponse {
  id: string;
  email: string;
  role: UserRole;
  is_used: boolean;
  created_at: string;
  expires_at: string;
}
